models/models.go

```diff
@@ -157,11 +157,9 @@ func SetEngine() (err error) {
 	// so use log file to instead print to stdout.
 	x.SetLogger(NewXORMLogger(setting.Database.LogSQL))
 	x.ShowSQL(setting.Database.LogSQL)
-	if setting.Database.UseMySQL {
-		x.SetMaxIdleConns(setting.Database.MaxIdleConns)
-		x.SetConnMaxLifetime(setting.Database.ConnMaxLifetime)
-	}
-
+	x.SetMaxIdleConns(setting.Database.MaxIdleConns)
+	x.SetConnMaxLifetime(setting.Database.ConnMaxLifetime)
+	x.SetMaxOpenConns(setting.Database.MaxOpenConns)
```

Comment:
According to this source ([and docs](https://golang.org/pkg/database/sql/#DB.SetMaxIdleConns)):

https://github.com/golang/go/blob/3876bd67ef4a1972f54d926f7d8a5c7324861f51/src/database/sql/sql.go#L898-L909

... it should be desirable to call`SetMaxOpenConns()` before calling `SetMaxIdleConns()`; otherwise, the latter could be overwritten.

Reply:
good catch.